knowledge-base/inputs-validation-child-component.md

```diff
@@ -33,155 +33,187 @@ res_type: kb
 
 ## Description
 
-I am wrapping a Telerik component inside a custom component for my application. When I try to validate it the red invalid border does not appear.
-
+I am wrapping a Telerik component inside a custom component in my Form. When I try to validate it the red invalid border does not appear.
 
 ## Solution
 
-Internally, the Telerik Blazor components use the cascading `EditContext` parameter that the `EditForm` provides to determine if validation passes or fails. If it fails, we add a class that shows a red border around the component. 
-
-When you abstract the component in a custom component you should specify the `ValueExpression` - this is the field that notifies the framework (`EditForm`) what value should pass certain criteria. It is generated automatically by the framework when using `@bind-Value` when directly in the edit form, but not when there is another component in the hierarchy.
+Internally, the Telerik input components use the cascading `EditContext` parameter that the `EditForm` and `TelerikForm` provide. The `EditContext` API allow the components to determine if validation has passed or failed. If the validation fails, the components show a red border.
+
+When you wrap an input component to another component you must define `ValueExpression` parameter in the custom component. This will allow the custom component to receive the correct expression from the parent component which holds the Form. The Blazor framework generates the expression automatically when using `@bind-Value`, but not when there is another component in the component hierarchy tree.
```

Comment:
```suggestion
When you wrap an input component in another component, you must define a `ValueExpression` parameter in the custom component. This will allow the custom component to receive the correct expression from the parent component, which holds the Form. The Blazor framework generates the expression automatically when using `@bind-Value`, but not when there is another component in the component hierarchy tree.
```